handlers/handlers.go

```diff
@@ -246,11 +249,83 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content string `json:"content"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := &models.Note{
+			AuthorId:     models.UserId(userId),
+			Content:      noteForm.Content,
+			CreationTime: time.Now().UTC(),
+		}
+
+		noteId, err := noteservice.StoreNewNote(note)
+		if err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		responseWriter.Write([]byte(`{NoteId: "` + strconv.FormatInt(int64(noteId), 10) + `"}`))
+		responseWriter.WriteHeader(http.StatusCreated)
+
 	default:
-		respondWithMethodNotAllowed(responseWriter, http.MethodGet)
+		respondWithMethodNotAllowed(responseWriter, http.MethodGet, http.MethodPost)
 	}
 }
 
+func HandleCategoryApiRequest(
```

Comment:
should specify `NoteCategory`

Reply:
You want it to read `HandleNoteCateogryApiRequest`?

Reply:
yeah that seems reasonable. Just saying `Category` seems a bit confusing